sql/src/main/scala/org/apache/spark/sql/sedona_sql/io/GeotiffFileFormat.scala

```diff
@@ -101,4 +129,127 @@ private[spark] class GeotiffFileFormat extends FileFormat with DataSourceRegiste
       }
     }
   }
+
+  private def isValidGeoTiffSchema(imageWriteOptions: ImageWriteOptions, dataSchema: StructType): Boolean = {
+    val fields = dataSchema.fieldNames
+    if (fields.contains(imageWriteOptions.colImage) ){
+      val schemaFields = dataSchema.fields(dataSchema.fieldIndex(imageWriteOptions.colImage)).dataType.asInstanceOf[StructType]
+      if (schemaFields.fieldNames.length != 6) return false
+    }
+    else {
+      if (fields.length != 6) return false
+    }
+    true
+  }
+
+}
+
+// class for writing geoTiff images
+private class GeotiffFileWriter(savePath: String,
+                                imageWriteOptions: ImageWriteOptions,
+                                dataSchema: StructType,
+                                context: TaskAttemptContext) extends OutputWriter {
+
+  // set writing parameters
+  private val DEFAULT_WRITE_PARAMS: GeoTiffWriteParams = new GeoTiffWriteParams()
+  DEFAULT_WRITE_PARAMS.setCompressionMode(ImageWriteParam.MODE_EXPLICIT)
+  DEFAULT_WRITE_PARAMS.setCompressionType("LZW")
+  DEFAULT_WRITE_PARAMS.setCompressionQuality(0.75F)
+  DEFAULT_WRITE_PARAMS.setTilingMode(ImageWriteParam.MODE_EXPLICIT)
+  DEFAULT_WRITE_PARAMS.setTiling(512, 512)
+
+  private val hfs = new Path(savePath).getFileSystem(context.getConfiguration)
+
+  override def write(row: InternalRow): Unit = {
+    // retrieving the metadata of a geotiff image
+    println("Schema:")
```

Comment:
Don't use println to print log info. Use Log4j instead. See the example: https://github.com/apache/incubator-sedona/blob/master/sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/Functions.scala#L1004

Use "info" level in logging. To see the log in the test, set log level to info level in TestBase.